### PR TITLE
8305427: Write a test Check whether focus changes as expected when requestFocus is used to traverse randomly around the window

### PR DESCRIPTION
**Pull request by ravigupta30**
This testcase Checking whether the Component becoming the Focus Owner and FocusEvent.FOCUS_GAINED will be received by the Component when the focus is requested on the component using requestFocus.

Testing:
Tested using Mach5(20 times per platform) in macos,linux and windows and got all pass.

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8305427](https://bugs.openjdk.org/browse/JDK-8305427): Write a test Check whether focus changes as expected when requestFocus is used to traverse randomly around the window (**Enhancement** - P4)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/13293/head:pull/13293` \
`$ git checkout pull/13293`

Update a local copy of the PR: \
`$ git checkout pull/13293` \
`$ git pull https://git.openjdk.org/jdk.git pull/13293/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 13293`

View PR using the GUI difftool: \
`$ git pr show -t 13293`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/13293.diff">https://git.openjdk.org/jdk/pull/13293.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/13293#issuecomment-1494064525)

### CODE REVIEW

**Comment by ravigupta30** [comment]
Any volunteers for a review?

**Review by DamonGuy — COMMENTED** [review]
Test passes on my local MacOS machine. 

Question: The title and bug description mentions testing requestFocus but this test uses requestFocusInWindow. Are you actually testing the right thing here?

Also are you covering enough UI components? I'm not sure the exact purpose of this test, but only CheckBox, Choice, TextField, and List are calling requestFocusInWindow. Button is being clicked by Robot instead. Then there's TextAreas and ScrollBar as other UI components to name a few.

File: test/jdk/java/awt/Focus/RequestFocusOwnerTest/RequestFocusOwnerTest.java [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+            }
+
+            focusGained = false;
+            EventQueue.invokeAndWait(() -> {
+                requestStatus = checkbox.requestFocusInWindow();
+            });
+
+            robot.waitForIdle();
+
+            if (!requestStatus) {
```
Comment (lawrence-andrew) [inline test/jdk/java/awt/Focus/RequestFocusOwnerTest/RequestFocusOwnerTest.java:144]: According to the requestFocusInWindow() documentation https://docs.oracle.com/en/java/javase/20/docs/api/java.desktop/java/awt/Component.html#requestFocusInWindow() . Developers must never assume that this Component is the focus owner until this Component receives a FOCUS_GAINED event.   

So dn't depend on requestFocusInWindow()  return value

**Comment by ravigupta30** [comment]
Any volunteers for a review?

**Comment by lawrence-andrew** [comment]
LGTM

**Comment by ravigupta30** [comment]
Any volunteers for a review?

**Review by aivanov-jdk — CHANGES_REQUESTED** [review]

**Pull request closed without merging**